lib/MySQL_Monitor.cpp

```diff
@@ -5816,5 +5891,1340 @@ bool DNS_Cache::empty() const {
+			process_ready_tasks_cb callback, MySQL_Monitor* mysql_monitor) :
+			min_task_to_process_(min_tasks_to_process), process_task_percentage_(percentage / 100.00), process_ready_tasks_cb_(callback),
+			mysql_monitor_(mysql_monitor) {
+			assert(mysql_monitor_);
+			assert(process_ready_tasks_cb_);
+			assert(process_task_percentage_ != 0);
+			assert(min_tasks_to_process != 0);
+		}
+		~Process_Ready_Task_Callback_Args() = default;
+
+	private:
+		inline
+		bool process_ready_tasks(const std::vector<MySQL_Monitor_State_Data*>& mmsds) {
+			return (mysql_monitor_->*process_ready_tasks_cb_)(mmsds);
+		}
+
+		friend class Monitor_Poll;
+		unsigned int min_task_to_process_;
+		float process_task_percentage_;
+		process_ready_tasks_cb process_ready_tasks_cb_;
+		MySQL_Monitor* mysql_monitor_;
+	};
+
+	Monitor_Poll(unsigned int capacity) {
+		len_ = 0;
+		capacity_ = capacity;
+		fds_ = (struct pollfd*)malloc(capacity_ * sizeof(struct pollfd));
+		mmsds_ = (MySQL_Monitor_State_Data**)malloc(capacity_ * sizeof(MySQL_Monitor_State_Data*));
+	}
+
+	~Monitor_Poll() {
+		free(fds_);
+		free(mmsds_);
+	}
+
+	void expand(unsigned int more) {
+		if ((len_ + more) > capacity_) {
+			unsigned int new_size = near_pow_2(len_ + more);
+			fds_ = (struct pollfd*)realloc(fds_, new_size * sizeof(struct pollfd));
+			mmsds_ = (MySQL_Monitor_State_Data**)realloc(mmsds_, new_size * sizeof(MySQL_Monitor_State_Data*));
+			capacity_ = new_size;
+		}
+	}
+
+	void add(short _events, MySQL_Monitor_State_Data* mmsd) {
+		assert(mmsd);
+		assert(mmsd->mysql);
+
+		if (len_ == capacity_) {
+			expand(1);
+		}
+		fds_[len_].fd = mysql_get_socket(mmsd->mysql);
+		fds_[len_].events = _events;
+		fds_[len_].revents = 0;
+		mmsds_[len_] = mmsd;
+		len_++;
+
+		mmsd->init_async();
+		mmsd->task_handler(-1, _events);
+	}
+
+	void remove_index_fast(unsigned int i) {
+		if ((int)i == -1) return;
+
+		if (i != (len_ - 1)) {
+			fds_[i].fd = fds_[len_ - 1].fd;
+			fds_[i].events = fds_[len_ - 1].events;
+			fds_[i].revents = fds_[len_ - 1].revents;
+			mmsds_[i] = mmsds_[len_ - 1];
+		}
+		len_--;
+	}
+
+	int find_index(int fd) {
+		unsigned int i;
+		for (i = 0; i < len_; i++) {
+			if (fds_[i].fd == fd) {
+				return i;
+			}
+		}
+		return -1;
+	}
+
+	bool event_loop(int poll_timeout_ms, Process_Ready_Task_Callback_Args& process_ready_task_callback_arg) {
+
+		if (len_ == 0)
+			return false;
+
+		int rc = 0;
+
+		// number of tasks to process based on provided percentage
+		unsigned int tasks_to_process_count = len_ * process_ready_task_callback_arg.process_task_percentage_;
+
+		// if number of task to process is less than minimum task to process, overwrite it
+		if (tasks_to_process_count < process_ready_task_callback_arg.min_task_to_process_) {
+			tasks_to_process_count = process_ready_task_callback_arg.min_task_to_process_;
+		}
+
+		std::vector<MySQL_Monitor_State_Data*> ready_tasks;
+		ready_tasks.reserve(tasks_to_process_count);
+
+		while (len_) {
+
```

Comment:
Empty line as mentioned in other comments.

Reply:
It's done intentionally to avoid code clutter.